### DISCUSSION TITLE
How do I get the ECS task metadata endpoint (environment variable)?

### DISCUSSION THREAD

**Discussion by sirbully** [aws/copilot-cli#5207] [Q&A] [answered]
It says on the [docs](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-metadata-endpoint-v4-fargate.html) that:

> Beginning with Fargate platform version `1.4.0`, an environment variable named `ECS_CONTAINER_METADATA_URI_V4` is injected into each container in a task.

These are the only environment variables available:

<img width="1058" alt="Screenshot 2023-08-21 at 0 50 07" src="https://github.com/aws/copilot-cli/assets/29673385/cca07d9d-ec85-44a5-9d10-304b980f0b7c">

Am I misunderstanding something?

**Comment by dannyrandall** [comment, marked as answer]
Hey @sirbully! `ECS_CONTAINER_METADATA_URI_V4` is injected into your container, but the only ones listed in the view you're looking at are variables specified by you in the manifest. If you want to get the value of `ECS_CONTAINER_METADATA_URI_V4` outside of your container, you could try using `copilot svc exec`, and then running `echo $ECS_CONTAINER_METADATA_URI_V4`!
